doc/install/mtls-configuration.md

```diff
@@ -0,0 +1,150 @@
+# Configure mTLS between the Gateway and Kuadrant components
+
+## Overview
+
+This guide includes steps to enable mTLS between the gateway and Kuadrant components.
+If you use an AuthPolicy or RateLimitPolicy, there will be communication between the gateway and the respective Kuadrant components at request time. This communication happens between the Wasm plugin in Envoy proxy, and Authorino or Limitador.
+
+!!! note
+
+    This method currently only works if the Gateway is provided by Istio.
+
+## Prerequisites
+
+You have installed Kuadrant in a [kubernetes](https://docs.kuadrant.io/latest/kuadrant-operator/doc/install/install-kubernetes/) or [OpenShift](https://docs.kuadrant.io/latest/kuadrant-operator/doc/install/install-openshift/) cluster.
+
+## Enabling mTLS
+
+### Kuadrant components
+
+As the Kuadrant components (Authorino & Limitador) are already part of the service mesh in Istio, mTLS can be enabled after an Envoy proxy sidecar is deployed along side them.
+To do this, apply the Istio sidecar label to both Deployments templates.
+
+```bash
+kubectl -n kuadrant-system patch deployment authorino \
+  -p '{"spec":{"template":{"metadata":{"labels":{"sidecar.istio.io/inject":"true"}}}}}'
+
+kubectl -n kuadrant-system patch deployment limitador-limitador \
+  -p '{"spec":{"template":{"metadata":{"labels":{"sidecar.istio.io/inject":"true"}}}}}'
+```
+
+You should see the number of containers in either pod increase from 1 to 2, as the `istio-proxy` is added to the pods. This change will force all traffic to those pods to go through the proxy. However, mTLS is not enabled yet.
+
+### Envoy Filter
+
+The next step enables mTLS for traffic originating in the gateway (where the Wasm plugin executes), going to the Kuadrant components.
+This requires modifying the EnvoyFilters directly.
```

Comment:
I'm actually surprised this works. In theory, any events related to the following CRs should reconcile the EnvoyFilter back to no-TLS form:
- Kuadrant
- GatewayClass
- Gateway
- HTTPRoute
- AuthPolicy
- EnvoyFilter

Reply:
I'll add a caveat that it may get undone, and they should allow for that by having some job or gitops thing reconcile it back.
I'll also link to the RFC where they can engage on it potential being a feature of kuadrant.